vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/HttpRequestStrategy.java

```diff
@@ -306,4 +323,51 @@ public void destroy() {
         }
     }
 
+    /**
+     * Oof, this is an attempt to see if there's a terminal bug in the Jetty client library that sometimes
+     * renders a client instance permanently unusable. If this is the case, replacing the client altogether
+     * should allow the feeder to start working again, when it wouldn't otherwise be able to.
+     */
+    private static class ResettableCluster implements Cluster {
+
+        private final Object monitor = new Object();
+        private final Deque<CompletableFuture<?>> inflight = new ArrayDeque<>();
+        private final Supplier<Cluster> delegates;
+        private Cluster delegate;
+
+        ResettableCluster(Supplier<Cluster> delegates) {
+            this.delegates = delegates;
+            this.delegate = delegates.get();
+        }
+
+        @Override
+        public void dispatch(HttpRequest request, CompletableFuture<HttpResponse> vessel) {
+            inflight.add(vessel);
```

Comment:
`inflight` will hold a reference to all previous and current operations until a reset occur? So it will leak memory until the circuit breaker is tripped?